h2o-genmodel/src/main/java/hex/genmodel/algos/tree/SharedTreeMojoReader.java

```diff
@@ -22,7 +22,38 @@ protected void readModelData() throws IOException {
     _model._ntrees_per_group = tpc;
     _model._compressed_trees = new byte[_model._ntree_groups * tpc][];
     _model._mojo_version = ((Number) readkv("mojo_version")).doubleValue();
-
+    _model._genmodel_encoding = readkv("_genmodel_encoding").toString();
+    if (_model._mojo_version < 1.40 && _model._genmodel_encoding != null && !"AUTO".equals(_model._genmodel_encoding)) {
+        throw new RuntimeException("Categorical encoding " + _model._genmodel_encoding + " is not supported in this version of MOJO. Please use version 1.4 or higher.");
+    }
+    if (_model._mojo_version >= 1.40) {
+        _model._orig_projection_array = readkv("_orig_projection_array");
+        Integer n = readkv("_n_orig_names");
+        if (n != null) {
+            _model._orig_names = new String[n];
+            int i = 0;
+            for (String line : readtext("_orig_names")) {
+                _model._orig_names[i++] = line;
+            }
+        }
+        n = readkv("_n_orig_domain_values");
+        if (n != null) {
+            _model._orig_domain_values = new String[n][];
+            for (int i = 0; i < n; i++) {
+                int m = readkv("_m_orig_domain_values_" + i);
+                if (m > 0) {
+                    String[] orig_domain_values_row = new String[m];
```

Comment:
I would also refactor reading of the String array into a reusable method

Reply:
fixed